index.js

```diff
@@ -14,44 +13,89 @@ function createResolver(config, options) {
     resolve: resolve,
   };
 
-  var lastKey;
 
-  function resolve(key) {
-    if (typeof key !== 'string') {
+  // Keep constants separately
+  var constants = {};
+
+  function resolveConstant(key) {
+    if (constants.hasOwnProperty(key)) {
+      return constants[key];
+    }
+
+    var definition = config[key];
+    // Ignore options that are not defined
+    if (!definition) {
       return;
     }
 
-    var appliedArgs = slice.call(arguments, 1);
+    var option = options[key];
+
+    if (!!option || options.hasOwnProperty(key)) {
```

Comment:
It'd have to be `option != null`, if I understand you right, and would not catch `false` as an explicitly set value, when it should be treated as a constant... The thought of this test was to catch the common case of a truthy value without further function call and to catch explicit falsy values with `hasOwnProperty`. Make sense?